### ISSUE TITLE
Empty array become null in json output after with_entries(select(...))

### ISSUE THREAD

**Issue by linyinfeng** [mikefarah/yq#1880]
**Describe the bug**

Version of yq: 4.40.2
Operating system: linux
Installed via: binary release

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)
data.yml:
```yaml
a: []
```

**Command**
The command you ran:
```
yq eval 'with_entries(select(.key == "a"))' data.yaml --output-format json
```

**Actual behavior**

```json
{
  "a": null
}
```

**Expected behavior**

```json
{
  "a": []
}
```

**Comment by mikefarah** [comment]
Hey thanks for raising! That's a nasty bug - will make a new release shortly.

**Comment by mikefarah** [comment]
Fixed in 4.40.3
